internal/gcsx/file_cache_reader_test.go

```diff
@@ -81,3 +98,209 @@ func (t *FileCacheReaderTest) TestNewFileCacheReader() {
 	assert.Nil(t.T(), reader.metricHandle)
 	assert.Nil(t.T(), reader.fileCacheHandle)
 }
+
+func (t *fileCacheReaderTest) Test_ReadAt_NilFileCacheHandlerThrowFallBackError() {
+	reader := NewFileCacheReader(t.object, t.mockBucket, nil, true, nil)
+
+	readerResponse, err := reader.ReadAt(t.ctx, make([]byte, 10), 0)
+
+	assert.True(t.T(), errors.Is(err, FallbackToAnotherReader))
+	assert.Zero(t.T(), readerResponse.Size)
+}
+
+// Writing unit tests on tryReadingFromFileCache to check if cache hit is getting populated correctly.
```

Comment:
nit: Writing unit tests? - we can remove this, test-method name conveys everything.

Reply:
Generally, it is preferable to write unit tests for public methods rather than helper functions. To avoid confusion for the reader, I've kept this.